api/src/auth.rs

```diff
@@ -178,7 +181,7 @@ fn parse_token(data: &[u8]) -> Result<Token, Response>
     let json = std::str::from_utf8(data)
         .map_err(|_| Response::new(StatusCode::InternalServerError))?;
     let toky = serde_json::from_str::<Token>(json)
-        .map_err(|_| Response::new(StatusCode::InternalServerError))?;
+        .map_err(|_| Response::new(StatusCode::Unauthorized))?;
```

Comment:
I'm not sure if the server failing to parse json from a string constitutes unauthorized. The client may have provided a correct authorization header and we just failed on our side. What is your thinking on this one?

Reply:
I went and back and forth on this a little. We have a Token structure that I added a .blake3 member to, there are cases when a request may be in process and find an old token which does not have this member. So the logic i actually want is: does this token have the blake3 field? if so then it's an actual error, if not then we should return Unauthorized so that the client will try again (or do something smart).

As far as i can tell, and i might be missing something, the only requests that are affected are those that are currently in process, once this deploy happens, any new requests will have the correctly formatted token.

Reply:
originally i had a comment elsewhere that said the best solution to this was to have two types of Token structures, but i think it got erased along with other changes I made. And also in the case of the two structures, like above, i think it only helps us for a small edge case

Reply:
I guess I was thinking more of the "spurious error" scenario where somehow we get garbled json. That case seems more clearly like an internal server error. I understand the in-flight request migration issue and agree it's not a big deal. I think a client would retry in either case but from the client side I think a 500 is a good hint that that the issue is transient in nature and retrying is appropriate whereas an unauthorized makes me think that my request was successful and I simply don't have access at the moment. I don't think it's a big deal either way though I think I lean towards 500 after pondering things more.

Reply:
500 is more correct i think as well

Reply:
@ceberly since we agree on this want to update quick and then I'll approve?

Reply:
this change was made